packages/datetime/src/timePicker.tsx

```diff
@@ -252,10 +274,30 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         this.updateTime(loopTime(newTime, unit), unit);
     }
 
+    private isTimeInRange(date: Date) {
+        const format = "H:mm:ss:SSS";
+        const time = moment(this.getDateOnlyWithTime(date), format);
+        const minTime = moment(this.getDateOnlyWithTime(this.props.minTime), format);
+        const maxTime = moment(this.getDateOnlyWithTime(this.props.maxTime), format);
+
+        const isTimeAfterMinTime = time.isSameOrAfter(minTime);
+        const isTimeBeforeMaxTime = time.isSameOrBefore(maxTime);
+
+        if (maxTime.isBefore(minTime)) {
+            return isTimeAfterMinTime || isTimeBeforeMaxTime;
+        }
+
+        return isTimeAfterMinTime && isTimeBeforeMaxTime;
+    }
+
+    private getDateOnlyWithTime(date: Date): Date {
+        return new Date(0, 0, 0, date.getHours(), date.getMinutes(), date.getSeconds(), date.getMilliseconds());
+    }
+
     private updateTime(time: number, unit: TimeUnit) {
-        if (isTimeValid(time, unit)) {
-            const newValue = DateUtils.clone(this.state.value);
-            setTimeUnit(time, newValue, unit);
+        const newValue = DateUtils.clone(this.state.value);
+        setTimeUnit(time, newValue, unit);
```

Comment:
Do you think it's safe to call `setTimUnit()` before validating input values with `isTimeValid()`?